--- a/atom_pubsub/src/atom_microblog.erl
+++ b/atom_pubsub/src/atom_microblog.erl
@@ -7,8 +7,9 @@
 -author('eric@ohmforce.com').
 -include("ejabberd.hrl").
 -include("jlib.hrl").
--include("mod_pubsub/pubsub.hrl").
--include("web/ejabberd_http.hrl").
+-include("pubsub.hrl").
+-include("ejabberd_http.hrl").
+-include("logger.hrl").
 -export([process/2]).
  
 process([Domain,User|_]=LocalPath,  #request{auth = Auth} = Request)->
--- a/atom_pubsub/src/atom_pubsub.erl
+++ b/atom_pubsub/src/atom_pubsub.erl
@@ -7,8 +7,9 @@
 -author('eric@ohmforce.com').
 -include("ejabberd.hrl").
 -include("jlib.hrl").
--include("mod_pubsub/pubsub.hrl").
--include("web/ejabberd_http.hrl").
+-include("pubsub.hrl").
+-include("ejabberd_http.hrl").
+-include("logger.hrl").
 -export([process/2]).
  
 process([Domain,User|_]=LocalPath,  #request{auth = Auth} = Request)->
