--- a/mod_admin_extra/src/mod_admin_extra.erl
+++ b/mod_admin_extra/src/mod_admin_extra.erl
@@ -164,7 +164,7 @@
      #ejabberd_commands{name = export2odbc, tags = [mnesia], %% Copied to ejabberd 2.1.x after 11
 			desc = "Export Mnesia tables to files in directory",
 			module = ?MODULE, function = export2odbc,
-			args = [{host, string}, {path, string}],
+			args = [{host, binary}, {path, string}],
 			result = {res, rescode}},
 
      #ejabberd_commands{name = num_active_users, tags = [accounts, stats],
@@ -241,8 +241,8 @@
 			args = [{host, binary}, {status, binary}],
 			result = {users, {list,
 					  {userstatus, {tuple, [
-								{user, string},
-								{host, string},
+								{user, binary},
+								{host, binary},
 								{resource, string},
 								{priority, integer},
 								{status, string}
@@ -254,8 +254,8 @@
 			args = [{status, binary}],
 			result = {users, {list,
 					  {userstatus, {tuple, [
-								{user, string},
-								{host, string},
+								{user, binary},
+								{host, binary},
 								{resource, string},
 								{priority, integer},
 								{status, string}
@@ -282,7 +282,7 @@
                        tags = [session],
                        desc = "Get the list of established sessions in a vhost",
                        module = ?MODULE, function = connected_users_vhost,
-                       args = [{host, string}],
+                       args = [{host, binary}],
                        result = {connected_users_vhost, {list, {sessions, string}}}},
      #ejabberd_commands{name = user_sessions_info,
 			tags = [session],
@@ -308,7 +308,7 @@
 			tags = [session],
 			desc = "Set presence of a session",
 			module = ?MODULE, function = set_presence,
-			args = [{user, string}, {host, string},
+			args = [{user, binary}, {host, binary},
 				{resource, string}, {type, string},
 				{show, string}, {status, string},
 				{priority, string}],
@@ -317,7 +317,7 @@
      #ejabberd_commands{name = set_nickname, tags = [vcard],
 			desc = "Set nickname in a user's vCard",
 			module = ?MODULE, function = set_nickname,
-			args = [{user, string}, {host, string}, {nickname, string}],
+			args = [{user, binary}, {host, binary}, {nickname, string}],
 			result = {res, rescode}},
 
      #ejabberd_commands{name = get_vcard, tags = [vcard],
@@ -410,7 +410,7 @@
 			result = {response,
 				  {list,
 				   {pairs, {tuple,
-					       [{user, string},
+					       [{user, binary},
 						{contact, string}
 					       ]}}
 				  }}},
@@ -428,7 +428,7 @@
      #ejabberd_commands{name = push_roster, tags = [roster],
 			desc = "Push template roster from file to a user",
 			module = ?MODULE, function = push_roster,
-			args = [{file, string}, {user, string}, {host, string}],
+			args = [{file, string}, {user, binary}, {host, binary}],
 			result = {res, rescode}},
      #ejabberd_commands{name = push_roster_all, tags = [roster],
 			desc = "Push template roster from file to all those users",
@@ -438,7 +438,7 @@
      #ejabberd_commands{name = push_alltoall, tags = [roster],
 			desc = "Add all the users to all the users of Host in Group",
 			module = ?MODULE, function = push_alltoall,
-			args = [{host, string}, {group, string}],
+			args = [{host, binary}, {group, string}],
 			result = {res, rescode}},
 
      #ejabberd_commands{name = get_last, tags = [last],
@@ -453,7 +453,7 @@
 			longdesc = "Timestamp is the seconds since"
 			"1970-01-01 00:00:00 UTC, for example: date +%s",
 			module = ?MODULE, function = set_last,
-			args = [{user, string}, {host, string}, {timestamp, integer}, {status, string}],
+			args = [{user, binary}, {host, binary}, {timestamp, integer}, {status, binary}],
 			result = {res, rescode}},
 
      #ejabberd_commands{name = private_get, tags = [private],
