--- a/mod_archive/src/mod_archive_odbc.erl
+++ b/mod_archive/src/mod_archive_odbc.erl
@@ -84,6 +84,7 @@
 
 -include("ejabberd.hrl").
 -include("jlib.hrl").
+-include("logger.hrl").
 
 -record(state, {host,
                 sessions,
--- a/mod_archive/src/mod_archive_webview.erl
+++ b/mod_archive/src/mod_archive_webview.erl
@@ -15,8 +15,9 @@
 
 -include("ejabberd.hrl").
 -include("jlib.hrl").
--include("web/ejabberd_http.hrl").
--include("web/ejabberd_web_admin.hrl"). %for all the defines
+-include("logger.hrl").
+-include("ejabberd_http.hrl").
+-include("ejabberd_web_admin.hrl"). %for all the defines
 
 -define(LINK(L) , "/archive/" ++ L).
 %-define(P(Els), ?XE("p", Els)).
