--- a/mod_admin_extra/src/mod_admin_extra.erl
+++ b/mod_admin_extra/src/mod_admin_extra.erl
@@ -694,7 +694,7 @@
 		    %% If it isnt
 		    [] ->
 			%% Look for his last_activity
-			case (get_lastactivity_module(LServer)):get_last_info(LUser, LServer) of
+			case mod_last:get_last_info(LUser, LServer) of
 			    %% If it is
 			    %% existent:
 			    {ok, TimeStamp, _Status} ->
@@ -728,12 +728,6 @@
     Users_removed = lists:filter(F, Users),
     {removed, length(Users_removed), Users_removed}.
 
-get_lastactivity_module(Server) ->
-    case lists:member(mod_last, gen_mod:loaded_modules(Server)) of
-        true -> mod_last;
-        _ -> mod_last_odbc
-    end.
-
 
 %%
 %% Ban account
@@ -1221,10 +1215,9 @@
 %%%
 
 get_last(User, Server) ->
-    Mod = get_lastactivity_module(Server),
     case ejabberd_sm:get_user_resources(User, Server) of
         [] ->
-            case Mod:get_last_info(User, Server) of
+            case mod_last:get_last_info(User, Server) of
                 not_found ->
                     "Never";
                 {ok, Shift, _Status} ->
@@ -1243,8 +1236,7 @@
     end.
 
 set_last(User, Server, Timestamp, Status) ->
-    Mod = get_lastactivity_module(Server),
-    Mod:store_last_info(User, Server, Timestamp, Status).
+    mod_last:store_last_info(User, Server, Timestamp, Status).
 
 %%%
 %%% Private Storage
