--- a/mod_webpresence/Emakefile
+++ b/mod_webpresence/Emakefile
@@ -1,2 +1,4 @@
-{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/mod_webpresence',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/mod_webpresence',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
--- a/atom_pubsub/Emakefile
+++ b/atom_pubsub/Emakefile
@@ -1,4 +1,6 @@
-{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/atom_microblog',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/atom_pubsub',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/mod_couch',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/atom_microblog',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/atom_pubsub',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/mod_couch',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
--- a/ircd/Emakefile
+++ b/ircd/Emakefile
@@ -1 +1,3 @@
-{'src/ejabberd_ircd',	[{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/ejabberd_ircd',	[{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
--- a/mod_archive/Emakefile
+++ b/mod_archive/Emakefile
@@ -1,5 +1,7 @@
-{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/mod_archive',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/mod_archive_sql',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/mod_archive_odbc',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/mod_archive_webview',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/mod_archive',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/mod_archive_sql',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/mod_archive_odbc',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/mod_archive_webview',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
--- a/mod_cron/Emakefile
+++ b/mod_cron/Emakefile
@@ -1,2 +1,4 @@
-{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/mod_cron',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/mod_cron',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
--- a/mod_log_chat/Emakefile
+++ b/mod_log_chat/Emakefile
@@ -1,4 +1,6 @@
-{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
-{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
-{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
 {'src/*',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
--- a/mod_muc_log_http/Emakefile
+++ b/mod_muc_log_http/Emakefile
@@ -1,2 +1,4 @@
-{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/mod_muc_log_http',     [{d,'SSL39'},{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/mod_muc_log_http',     [{d,'SSL39'},{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
--- a/mod_multicast/Emakefile
+++ b/mod_multicast/Emakefile
@@ -1,6 +1,8 @@
-{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/mod_multicast',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/ejabberd_router_multicast',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/ejabberd_c2s',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/mod_muc_room',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/ejabberd_sup',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/mod_multicast',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/ejabberd_router_multicast',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/ejabberd_c2s',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/mod_muc_room',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/ejabberd_sup',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
--- a/mod_profile/Emakefile
+++ b/mod_profile/Emakefile
@@ -1,2 +1,4 @@
-{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/mod_profile',	[{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/mod_profile',	[{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
--- a/ejabberd_auth_http/Emakefile
+++ b/ejabberd_auth_http/Emakefile
@@ -1,2 +1,4 @@
-{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/ejabberd_auth_http',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/ejabberd_auth_http',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
--- a/mod_post_log/Emakefile
+++ b/mod_post_log/Emakefile
@@ -1,2 +1,4 @@
-{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}]}.
-{'src/mod_post_log',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}]}.
+{'../ejabberd-dev/src/lager_transform', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/lager_util', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'../ejabberd-dev/src/gen_mod', [{outdir, "../ejabberd-dev/ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
+{'src/mod_post_log',     [{outdir, "ebin"},{i,"../ejabberd-dev/include"}, {d, 'LAGER'}]}.
