--- a/mod_cron/src/mod_cron.erl
+++ b/mod_cron/src/mod_cron.erl
@@ -25,7 +25,11 @@
 -include("ejabberd_http.hrl").
 -include("ejabberd_web_admin.hrl").
 -include("logger.hrl").
+-ifdef(NO_EXT_LIB).
 -include("xml.hrl").
+-else.
+-include_lib("p1_xml/include/xml.hrl").
+-endif.
 
 -record(task, {taskid, timerref, host, task}).
 
--- a/mod_log_chat/src/mod_log_chat.erl
+++ b/mod_log_chat/src/mod_log_chat.erl
@@ -16,7 +16,9 @@
 	 log_packet_send/3,
 	 log_packet_receive/4]).
 
+-ifndef(LAGER).
 -define(LAGER, 1).
+-endif.
 
 -include("ejabberd.hrl").
 -include("logger.hrl").
